src/lib/verifyAppleIdToken.ts

```diff
@@ -1,51 +1,56 @@
-import * as jwt from 'jsonwebtoken';
-import * as jwksClient from 'jwks-rsa';
+import * as jwt from 'jsonwebtoken'
+import * as jwksClient from 'jwks-rsa'
+import { VerifyAppleIdTokenParams, VerifyAppleIdTokenResponse } from './types'
 
-export const APPLE_BASE_URL = 'https://appleid.apple.com';
-export const JWKS_APPLE_URI = '/auth/keys';
+export const APPLE_BASE_URL = 'https://appleid.apple.com'
+export const JWKS_APPLE_URI = '/auth/keys'
 
-export interface VerifyAppleIdTokenParams {
-  idToken: string;
-  clientId: string | string[];
-  nonce?: string;
-}
-
-export const getApplePublicKey = async (kid) => {
+export const getApplePublicKey = async (kid: string) => {
   const client = jwksClient({
     cache: true,
     jwksUri: `${APPLE_BASE_URL}${JWKS_APPLE_URI}`,
-  });
-  const key: any = await new Promise((resolve, reject) => {
+  })
+  const key = await new Promise<jwksClient.SigningKey>((resolve, reject) => {
     client.getSigningKey(kid, (error, result) => {
       if (error) {
-        return reject(error);
+        return reject(error)
       }
-      return resolve(result);
-    });
-  });
-  return (key.publicKey as jwksClient.CertSigningKey) || (key.rsaPublicKey as jwksClient.RsaSigningKey);
-};
+      return resolve(result)
+    })
+  })
+  return key.getPublicKey()
+}
 
 export default async (params: VerifyAppleIdTokenParams) => {
-  const decoded = jwt.decode(params.idToken, { complete: true });
-  const { kid, alg } = decoded.header;
-  const applePublicKey = await getApplePublicKey(kid);
+  const decoded = jwt.decode(params.idToken, { complete: true })
+  const { kid, alg } = decoded.header
+
+  if (alg !== 'RS256') {
```

Comment:
Why do we throw error if the algo is `RS256`? Is it specified anywhere in the apple documentation or something like this? 

Just asking.

Reply:
removed and use assignment type